crates/core/c8y_api/src/smartrest/inventory.rs

```diff
@@ -116,6 +117,23 @@ pub fn service_creation_message_payload(
     ]))
 }
 
+/// Create a SmartREST message to set a response interval for c8y_RequiredAvailability.
+///
+/// The interval can be <=0, which means the device is in maintenance mode in the c8y context.
+/// Details: https://cumulocity.com/docs/device-integration/fragment-library/#device-availability
+pub fn set_required_availability_message(
+    c8y_topic: C8yTopic,
+    interval: i16,
+    prefix: &TopicPrefix,
+) -> MqttMessage {
```

Comment:
Rather than such an adhoc function, it would be better to `impl Into<MqttMessage> for C8ySmartRestSetInterval117`.

Reply:
Done in 2e05dc2b36fc1d10b9c6bcc28bd3e492e5520b05